### DISCUSSION TITLE
How to organize the example gallery?

### DISCUSSION THREAD

**Discussion by kandersolar** [pvlib/pvlib-python#1279] [General]
The [example gallery](https://pvlib-python.readthedocs.io/en/latest/auto_examples/index.html) now has 17 items and might benefit from being organized into sections.  This has been briefly discussed elsewhere:

- https://github.com/pvlib/pvlib-python/pull/1077#issuecomment-705690331
- https://github.com/pvlib/pvlib-python/pull/1266#issuecomment-892112481

I think the sphinx configuration changes to do this would be straightforward (see [here](https://sphinx-gallery.github.io/stable/configuration.html#sorting-gallery-subsections)).  The question is how the gallery should be organized.  For inspiration, here are some other packages with sections in their gallery:
- matplotlib: https://matplotlib.org/stable/gallery/index.html
- metpy: https://unidata.github.io/MetPy/latest/examples/index.html
- sunpy: https://docs.sunpy.org/en/stable/generated/gallery/index.html
- cartopy: https://scitools.org.uk/cartopy/docs/latest/gallery/index.html
- scikit-learn: https://scikit-learn.org/stable/auto_examples/index.html
- probably other ones listed here: https://sphinx-gallery.github.io/stable/projects_list.html

To get the conversation started, here are some categorizations mentioned in the above threads:
- by abstraction level: functions vs objects
- by task (or "how do I"), e.g. system modeling vs resource assessment vs ...
- by modeling topic, roughly along the lines of the pvlib module breakdown and PVPMC diagram

Input from anyone is welcome!

**Comment by mikofski** [comment]
I think I like by module? EG:
- solar position
- irradiance
- atmosphere
- clearsky
- PV modules
- inverters 
- IO
- IV curves
- PV systems
- etc

This could incorporate both topics and abstraction level, bc objects tend to aggregate several functions across topics. Each section could have both "how to's" and other less general examples

**Reply by kandersolar** [reply]
I'm not sure I like organizing by module, in part because some of these examples span multiple modules (`Modeling with interval averages`: solar position or irradiance?).  What do folks think about the proposed topic-based organization below?  It's mostly the same as module-based, but I think it might be clearer.  

- Transposition: [Mixed Orientation](https://pvlib-python.readthedocs.io/en/latest/auto_examples/plot_mixed_orientation.html), [GHI to POA Transposition](https://pvlib-python.readthedocs.io/en/latest/auto_examples/plot_ghi_transposition.html), [Seasonal Tilt](https://pvlib-python.readthedocs.io/en/latest/auto_examples/plot_seasonal_tilt.html), [Modeling Transposition Gain](https://pvlib-python.readthedocs.io/en/latest/auto_examples/plot_transposition_gain.html), [Modeling with interval averages](https://pvlib-python.readthedocs.io/en/latest/auto_examples/plot_interval_transposition_error.html)
- Tracking: [Single-axis tracking](https://pvlib-python.readthedocs.io/en/latest/auto_examples/plot_single_axis_tracking.html), [Dual-Axis Tracking](https://pvlib-python.readthedocs.io/en/latest/auto_examples/plot_dual_axis_tracking.html), [Backtracking on sloped terrain](https://pvlib-python.readthedocs.io/en/latest/auto_examples/plot_single_axis_tracking_on_sloped_terrain.html), [Discontinuous Tracking](https://pvlib-python.readthedocs.io/en/latest/auto_examples/plot_discontinuous_tracking.html)
- I-V Modeling: [Calculating a module's IV curves](https://pvlib-python.readthedocs.io/en/latest/auto_examples/plot_singlediode.html)
- Solar Position: [Sun path diagram](https://pvlib-python.readthedocs.io/en/latest/auto_examples/plot_sunpath_diagrams.html)
- Soiling: [Kimber Soiling Model](https://pvlib-python.readthedocs.io/en/latest/auto_examples/plot_greensboro_kimber_soiling.html), [HSU Soiling Model Example](https://pvlib-python.readthedocs.io/en/latest/auto_examples/plot_fig3A_hsu_soiling_example.html)
- Shading: [Diffuse Self Shading](https://pvlib-python.readthedocs.io/en/latest/auto_examples/plot_passias_diffuse_shading.html), [Calculating power loss from partial module shading](https://pvlib-python.readthedocs.io/en/latest/auto_examples/plot_partial_module_shading_simple.html)
- Reflections: [Diffuse IAM Calculation](https://pvlib-python.readthedocs.io/en/latest/auto_examples/plot_diffuse_aoi_correction.html)
- Spectrum: [Modeling Spectral Irradiance](https://pvlib-python.readthedocs.io/en/latest/auto_examples/plot_spectrl2_fig51A.html)
- End-to-end system modeling: (none yet, but I'd like to see a few examples of complete ModelChain workflows in here)
- ...

edit: moved `Modeling with interval averages` from solar position to transposition
